mapillary/mapillary.py

```diff
@@ -850,52 +856,64 @@ def image_from_key(key: int, fields: list = []) -> str:
 
 
 @auth()
-def save_to_csv(
-    csv_data,
-    file_path,
-):
-    """This function saves the csv data locally with the extension .csv
+def save_locally(
+    geojson_data: str,
+    file_path: str = os.path.dirname(os.path.realpath(__file__)),
+    file_name: str = None,
+    format: str = "geojson",
+) -> None:
+    """This function saves the geojson data locally as a file
+    with the given file name, path, and format.
 
-    :param csv_data: The CSV data to be stored
-    :type csv_data: CSV Object
+    :param geojson_data: The GeoJSON data to be stored
+    :type geojson_data: str
 
-    :param file_path: The path to save the data to
+    :param file_path: The path to save the data to. Defaults to the current directory path
     :type file_path: str
 
-    :return: None
-    :rtype: None
+    :param file_name: The name of the file to be saved. Defaults to 'geojson'
+    :type file_name: str
 
-    Usage::
-        # TODO: Write code here to display how
-        # TODO: the function works
-    """
+    :param format: The format to save the data as. Defaults to 'geojson'
+    :type format: str
 
-    # TODO: This functions needs implementation
+    Note: Allowed file format values at the moment are:
+        - geojson
+        - CSV
+    more file format values can be added and supported in the future
 
-    return None
-
-
-@auth()
-def save_as_geojson(
-    geojson_data,
-    file_path,
-):
-    """This function saves the geojson data locally with the extension .geojson
-
-    :param geojson_data: The GeoJSON data to be stored
-    :type geojson_data: GeoJSON Object
-
-    :param file_path: The path to save the data to
-    :type file_path: str
 
     :return: None
     :rtype: None
 
     Usage::
-        # TODO: Write code here to display how
-        # TODO: the function works
-    """
-
-    # TODO: This functions needs implementation
+        >>> import mapillary as mly
+        >>> mly.set_access_token('MLY|XXX')
+        >>> mly.save(
+        ...     geojson_data=geojson_data,
+        ...     file_path=os.path.dirname(os.path.realpath(__file__)),
+        ...     file_name='test_geojson',
```

Comment:
Maybe good to have a small check: if the user tries to put ".geojson" or ".csv" in the filename, then would save geojson.geojson.geojson. I've always wondered if that's worth throwing a warning or error to say the extension should not be in the file_name variable, but maybe it's okay to just leave it (the file will still open fine)

Reply:
Maybe the thing is to just prevent it from having a "." in the filename but still, that is more for the appearance than the function, so I wonder what you think

Reply:
@OmarMuhammedAli maybe we should put a check on what a user can name their files? Only ASCII characters, and what Chris suggested.

Reply:
Done